nomt/src/store.rs

```diff
@@ -95,13 +107,42 @@ pub struct Transaction {
 }
 
 impl Transaction {
-    /// Write a value to flat storage as part of the atomic batch.
-    /// Any previous value is overwritten. `None` means delete.
-    pub fn write_value(&mut self, path: KeyPath, value: Option<Vec<u8>>) {
-        let cf = self.shared.db.cf_handle(FLAT_KV_CF).unwrap();
+    /// Write a value to flat storage as well as its current leaf node representation
+    /// as part of the atomic batch. Provide the previous value's hash to clean up any previous leaf
+    /// node.
+    pub fn write_value<H: NodeHasher>(
```

Comment:
Would be good to note the differences of `Some` vs. `None` for `prev_value` and `value`.